generics: methods from struct aliases doesn't work

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.1 6bd3550
**OS:** linux, Ubuntu 20.04.1 LTS (VM) (WSL)

**What did you do?**
```v
struct Base {}
pub fn (base Base) hello() { println('hello') }
type Message = Base

fn hello<T>(hello_impl T) {	hello_impl.hello() }

fn main() {
	mut message := Message(Base{})
	hello(message)
	message.hello() // hello
}
```

**What did you expect to see?**
```v
hello
hello
```
 
**What did you see instead?**
```v
teris@DESKTOP-SQC5I5K:/mnt/c/Dev/v_feature$ v -cg run ./main.v
/tmp/v/main.15299100337987040392.tmp.c: In function ‘main__hello_T_main__Message’:
/tmp/v/main.15299100337987040392.tmp.c:8377:2: warning: implicit declaration of function ‘main__Message_hello’; did you mean ‘main__Base_hello’? [-Wimplicit-function-declaration]
 8377 |  main__Message_hello(hello_impl);
      |  ^~~~~~~~~~~~~~~~~~~
      |  main__Base_hello
/tmp/v/main.15299100337987040392.tmp.c: In function ‘main__main’:
/tmp/v/main.15299100337987040392.tmp.c:8383:1: warning: excess elements in struct initializer
 8383 | 0
      | ^
/tmp/v/main.15299100337987040392.tmp.c:8383:1: note: (near initialization for ‘(anonymous)’)
/usr/bin/ld: /tmp/ccQejpA5.o: in function `main__hello_T_main__Message':
/tmp/v/main.15299100337987040392.tmp.c:8377: undefined reference to `main__Message_hello'
collect2: error: ld returned 1 exit status
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -577,7 +577,12 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 			else {}
 		}
 	}
-	typ_sym := g.table.get_type_symbol(unwrapped_rec_type)
+	mut typ_sym := g.table.get_type_symbol(unwrapped_rec_type)
+	// alias type that undefined this method (not include `str`) need to use parent type
+	if typ_sym.kind == .alias && node.name != 'str' && !typ_sym.has_method(node.name) {
+		unwrapped_rec_type = (typ_sym.info as ast.Alias).parent_type
+		typ_sym = g.table.get_type_symbol(unwrapped_rec_type)
+	}
 	rec_cc_type := g.cc_type(unwrapped_rec_type, false)
 	mut receiver_type_name := util.no_dots(rec_cc_type)
 	if typ_sym.kind == .interface_ && (typ_sym.info as ast.Interface).defines_method(node.name) {

--- a/vlib/v/tests/generics_method_on_alias_struct_receiver_test.v
+++ b/vlib/v/tests/generics_method_on_alias_struct_receiver_test.v
@@ -1,0 +1,22 @@
+struct Base {}
+
+pub fn (base Base) hello() string {
+	return 'hello'
+}
+
+type Message = Base
+
+fn hello<T>(hello_impl T) string {
+	return hello_impl.hello()
+}
+
+fn test_generic_method_on_alias_struct_receiver() {
+	mut message := Message(Base{})
+	ret1 := hello(message)
+	println(ret1)
+	assert ret1 == 'hello'
+
+	ret2 := message.hello()
+	println(ret2)
+	assert ret2 == 'hello'
+}

